[Automation] java run fails for io.ktor:ktor-http-cio-jvm:3.4.1

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.4.1" ./gradlew clean javaTest -Pcoordinates="io.ktor:ktor-http-cio-jvm:3.2.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25541257498/job/74967784642
Last 300 lines of the log:
```console
* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 3s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.ktor:ktor-http-cio-jvm:3.2.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 5s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.4.1 at 2026-05-08T06:58:57Z with exit code 1 after 5s
❌ ERROR: 'native-image run' for 3.4.1 failed with exit code 1.
SOFT-FAIL[native-image run][3.4.1][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.4.1" ./gradlew clean test -Pcoordinates="io.ktor:ktor-http-cio-jvm:3.2.0"]
========================================================================================
 io.ktor:ktor-http-cio-jvm:3.2.0:3.4.1 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.4.1 at 2026-05-08T06:58:57Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.4.1" ./gradlew clean compileTestJava -Pcoordinates="io.ktor:ktor-http-cio-jvm:3.2.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 637ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileKotlin NO-SOURCE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar

> Task :compileTestKotlin
w: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt:308:39 'fun nameAt(idx: Int): CharSequence' is deprecated. Use nameAtOffset instead.
w: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt:308:62 'fun valueAt(idx: Int): CharSequence' is deprecated. Use valueAtOffset instead.

> Task :compileTestJava NO-SOURCE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
11 actionable tasks: 2 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.4.1 at 2026-05-08T06:59:00Z with exit code 0 after 3s
========================================================================================
 io.ktor:ktor-http-cio-jvm:3.2.0:3.4.1 stage java run
========================================================================================
Starting stage 'java run' for 3.4.1 at 2026-05-08T06:59:00Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.4.1" ./gradlew clean javaTest -Pcoordinates="io.ktor:ktor-http-cio-jvm:3.2.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 639ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :compileKotlin NO-SOURCE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE

> Task :compileTestKotlin
w: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt:308:39 'fun nameAt(idx: Int): CharSequence' is deprecated. Use nameAtOffset instead.
w: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt:308:62 'fun valueAt(idx: Int): CharSequence' is deprecated. Use valueAtOffset instead.

> Task :compileTestJava NO-SOURCE
> Task :testClasses UP-TO-DATE

> Task :test FAILED

Ktor_http_cio_jvmTest > encodesDecodesChunkedBodiesAndParsesChunkedHttpBody() FAILED
    java.lang.AssertionError: 
    Expecting actual throwable to be an instance of:
      java.lang.NumberFormatException
    but was:
      java.io.IOException: Invalid chunk size character: 0x6e
    	at io.ktor.http.cio.ChunkedTransferEncodingKt.parseChunkSize(ChunkedTransferEncoding.kt:157)
    	at io.ktor.http.cio.ChunkedTransferEncodingKt.decodeChunked(ChunkedTransferEncoding.kt:65)
    	at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1$1$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:213)
    	...(40 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
        at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:215)
        at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invoke(Ktor_http_cio_jvmTest.kt)
        at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invoke(Ktor_http_cio_jvmTest.kt)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndspatched(Undispatched.kt:66)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturnIgnoreTimeout(Undispatched.kt:50)
        at kotlinx.coroutines.TimeoutKt.setupTimeout(Timeout.kt:149)
        at kotlinx.coroutines.TimeoutKt.withTimeout(Timeout.kt:44)
        at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:188)
        at _COROUTINE._BOUNDARY._(CoroutineDebugging.kt:42)
        at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:188)

        Caused by:
        java.lang.AssertionError: 
        Expecting actual throwable to be an instance of:

9 tests completed, 1 failed

FAILURE: Build failed with an exception.

          java.lang.NumberFormatException
        but was:
          java.io.IOException: Invalid chunk size character: 0x6e
        	at io.ktor.http.cio.ChunkedTransferEncodingKt.parseChunkSize(ChunkedTransferEncoding.kt:157)
        	at io.ktor.http.cio.ChunkedTransferEncodingKt.decodeChunked(ChunkedTransferEncoding.kt:65)
        	at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1$1$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:213)
        	...(40 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
            at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:215)
            at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invoke(Ktor_http_cio_jvmTest.kt)
            at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invoke(Ktor_http_cio_jvmTest.kt)
            at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndspatched(Undispatched.kt:66)
            at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturnIgnoreTimeout(Undispatched.kt:50)
            at kotlinx.coroutines.TimeoutKt.setupTimeout(Timeout.kt:149)
            at kotlinx.coroutines.TimeoutKt.withTimeout(Timeout.kt:44)
            at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:188)
            at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
            at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
            at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:263)
            at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:94)
            at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:70)
            at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
            at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:48)
            at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
            at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest.encodesDecodesChunkedBodiesAndParsesChunkedHttpBody(Ktor_http_cio_jvmTest.kt:187)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/build/reports/problems/problems-report.html
* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).


BUILD FAILED in 3s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 3.4.1 at 2026-05-08T06:59:04Z with exit code 1 after 4s
❌ ERROR: 'java run' for 3.4.1 failed with exit code 1.
FAILED[java run][3.4.1][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.4.1" ./gradlew clean javaTest -Pcoordinates="io.ktor:ktor-http-cio-jvm:3.2.0"]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.4.1" ./gradlew clean javaTest -Pcoordinates="io.ktor:ktor-http-cio-jvm:3.2.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25541257498/job/74967784590
Last 300 lines of the log:
```console
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
14 actionable tasks: 4 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.ktor:ktor-http-cio-jvm:3.2.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 6s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.4.1 at 2026-05-08T07:00:54Z with exit code 1 after 7s
❌ ERROR: 'native-image run' for 3.4.1 failed with exit code 1.
SOFT-FAIL[native-image run][3.4.1][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.4.1" ./gradlew clean test -Pcoordinates="io.ktor:ktor-http-cio-jvm:3.2.0"]
========================================================================================
 io.ktor:ktor-http-cio-jvm:3.2.0:3.4.1 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.4.1 at 2026-05-08T07:00:54Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.4.1" ./gradlew clean compileTestJava -Pcoordinates="io.ktor:ktor-http-cio-jvm:3.2.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 871ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileKotlin NO-SOURCE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar

> Task :compileTestKotlin
w: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt:308:39 'fun nameAt(idx: Int): CharSequence' is deprecated. Use nameAtOffset instead.
w: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt:308:62 'fun valueAt(idx: Int): CharSequence' is deprecated. Use valueAtOffset instead.

> Task :compileTestJava NO-SOURCE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
11 actionable tasks: 2 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.4.1 at 2026-05-08T07:00:57Z with exit code 0 after 3s
========================================================================================
 io.ktor:ktor-http-cio-jvm:3.2.0:3.4.1 stage java run
========================================================================================
Starting stage 'java run' for 3.4.1 at 2026-05-08T07:00:57Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.4.1" ./gradlew clean javaTest -Pcoordinates="io.ktor:ktor-http-cio-jvm:3.2.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 760ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :compileKotlin NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE

> Task :compileTestKotlin
w: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt:308:39 'fun nameAt(idx: Int): CharSequence' is deprecated. Use nameAtOffset instead.
w: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt:308:62 'fun valueAt(idx: Int): CharSequence' is deprecated. Use valueAtOffset instead.

> Task :compileTestJava NO-SOURCE
> Task :testClasses UP-TO-DATE

> Task :test FAILED

Ktor_http_cio_jvmTest > encodesDecodesChunkedBodiesAndParsesChunkedHttpBody() FAILED
    java.lang.AssertionError: 
    Expecting actual throwable to be an instance of:
      java.lang.NumberFormatException
    but was:
      java.io.IOException: Invalid chunk size character: 0x6e
    	at io.ktor.http.cio.ChunkedTransferEncodingKt.parseChunkSize(ChunkedTransferEncoding.kt:157)
    	at io.ktor.http.cio.ChunkedTransferEncodingKt.decodeChunked(ChunkedTransferEncoding.kt:65)
    	at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1$1$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:213)
    	...(40 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
        at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:215)
        at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invoke(Ktor_http_cio_jvmTest.kt)
        at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invoke(Ktor_http_cio_jvmTest.kt)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndspatched(Undispatched.kt:66)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturnIgnoreTimeout(Undispatched.kt:50)
        at kotlinx.coroutines.TimeoutKt.setupTimeout(Timeout.kt:149)
        at kotlinx.coroutines.TimeoutKt.withTimeout(Timeout.kt:44)
        at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:188)
        at _COROUTINE._BOUNDARY._(CoroutineDebugging.kt:42)
        at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:188)

        Caused by:
        java.lang.AssertionError: 
        Expecting actual throwable to be an instance of:
          java.lang.NumberFormatException
        but was:
          java.io.IOException: Invalid chunk size character: 0x6e
        	at io.ktor.http.cio.ChunkedTransferEncodingKt.parseChunkSize(ChunkedTransferEncoding.kt:157)
        	at io.ktor.http.cio.ChunkedTransferEncodingKt.decodeChunked(ChunkedTransferEncoding.kt:65)
        	at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1$1$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:213)
        	...(40 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
            at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:215)
            at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invoke(Ktor_http_cio_jvmTest.kt)
            at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invoke(Ktor_http_cio_jvmTest.kt)
            at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndspatched(Undispatched.kt:66)
            at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturnIgnoreTimeout(Undispatched.kt:50)
            at kotlinx.coroutines.TimeoutKt.setupTimeout(Timeout.kt:149)
            at kotlinx.coroutines.TimeoutKt.withTimeout(Timeout.kt:44)
            at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:188)
            at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
            at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
            at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:263)
            at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:94)
            at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:70)
            at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
            at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:48)
            at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
            at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest.encodesDecodesChunkedBodiesAndParsesChunkedHttpBody(Ktor_http_cio_jvmTest.kt:187)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date

9 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 3s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 5s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 3.4.1 at 2026-05-08T07:01:03Z with exit code 1 after 6s
❌ ERROR: 'java run' for 3.4.1 failed with exit code 1.
FAILED[java run][3.4.1][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.4.1" ./gradlew clean javaTest -Pcoordinates="io.ktor:ktor-http-cio-jvm:3.2.0"]
```

## java run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.4.1" ./gradlew clean javaTest -Pcoordinates="io.ktor:ktor-http-cio-jvm:3.2.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25541257498/job/74967784580
Last 300 lines of the log:
```console
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date

BUILD FAILED in 4s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.ktor:ktor-http-cio-jvm:3.2.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 6s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.4.1 at 2026-05-08T07:01:54Z with exit code 1 after 6s
❌ ERROR: 'native-image run' for 3.4.1 failed with exit code 1.
SOFT-FAIL[native-image run][3.4.1][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.4.1" ./gradlew clean test -Pcoordinates="io.ktor:ktor-http-cio-jvm:3.2.0"]
========================================================================================
 io.ktor:ktor-http-cio-jvm:3.2.0:3.4.1 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.4.1 at 2026-05-08T07:01:54Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.4.1" ./gradlew clean compileTestJava -Pcoordinates="io.ktor:ktor-http-cio-jvm:3.2.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 821ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :compileKotlin NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar

> Task :compileTestKotlin
w: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt:308:39 'fun nameAt(idx: Int): CharSequence' is deprecated. Use nameAtOffset instead.
w: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt:308:62 'fun valueAt(idx: Int): CharSequence' is deprecated. Use valueAtOffset instead.

> Task :compileTestJava NO-SOURCE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
11 actionable tasks: 2 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.4.1 at 2026-05-08T07:01:58Z with exit code 0 after 4s
========================================================================================
 io.ktor:ktor-http-cio-jvm:3.2.0:3.4.1 stage java run
========================================================================================
Starting stage 'java run' for 3.4.1 at 2026-05-08T07:01:58Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.4.1" ./gradlew clean javaTest -Pcoordinates="io.ktor:ktor-http-cio-jvm:3.2.0"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 749ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :compileKotlin NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE

> Task :compileTestKotlin
w: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt:308:39 'fun nameAt(idx: Int): CharSequence' is deprecated. Use nameAtOffset instead.
w: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt:308:62 'fun valueAt(idx: Int): CharSequence' is deprecated. Use valueAtOffset instead.

> Task :compileTestJava NO-SOURCE
> Task :testClasses UP-TO-DATE

> Task :test FAILED

Ktor_http_cio_jvmTest > encodesDecodesChunkedBodiesAndParsesChunkedHttpBody() FAILED
    java.lang.AssertionError: 
    Expecting actual throwable to be an instance of:
      java.lang.NumberFormatException
    but was:
      java.io.IOException: Invalid chunk size character: 0x6e
    	at io.ktor.http.cio.ChunkedTransferEncodingKt.parseChunkSize(ChunkedTransferEncoding.kt:157)
    	at io.ktor.http.cio.ChunkedTransferEncodingKt.decodeChunked(ChunkedTransferEncoding.kt:65)
    	at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1$1$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:213)
    	...(40 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
        at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:215)
        at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invoke(Ktor_http_cio_jvmTest.kt)
        at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invoke(Ktor_http_cio_jvmTest.kt)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndspatched(Undispatched.kt:66)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturnIgnoreTimeout(Undispatched.kt:50)
        at kotlinx.coroutines.TimeoutKt.setupTimeout(Timeout.kt:149)
        at kotlinx.coroutines.TimeoutKt.withTimeout(Timeout.kt:44)
        at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:188)
        at _COROUTINE._BOUNDARY._(CoroutineDebugging.kt:42)
        at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:188)

        Caused by:
        java.lang.AssertionError: 
        Expecting actual throwable to be an instance of:
          java.lang.NumberFormatException
        but was:
          java.io.IOException: Invalid chunk size character: 0x6e
        	at io.ktor.http.cio.ChunkedTransferEncodingKt.parseChunkSize(ChunkedTransferEncoding.kt:157)
        	at io.ktor.http.cio.ChunkedTransferEncodingKt.decodeChunked(ChunkedTransferEncoding.kt:65)
        	at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1$1$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:213)
        	...(40 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
            at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:215)
            at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invoke(Ktor_http_cio_jvmTest.kt)
            at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1$1.invoke(Ktor_http_cio_jvmTest.kt)
            at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndspatched(Undispatched.kt:66)
            at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturnIgnoreTimeout(Undispatched.kt:50)
            at kotlinx.coroutines.TimeoutKt.setupTimeout(Timeout.kt:149)
            at kotlinx.coroutines.TimeoutKt.withTimeout(Timeout.kt:44)
            at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest$encodesDecodesChunkedBodiesAndParsesChunkedHttpBody$1.invokeSuspend(Ktor_http_cio_jvmTest.kt:188)
            at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
            at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
            at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:263)
            at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:94)
            at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:70)
            at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
            at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:48)
            at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
            at io_ktor.ktor_http_cio_jvm.Ktor_http_cio_jvmTest.encodesDecodesChunkedBodiesAndParsesChunkedHttpBody(Ktor_http_cio_jvmTest.kt:187)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/build/reports/problems/problems-report.html


9 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.ktor/ktor-http-cio-jvm/3.2.0/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 3s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 5s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 3.4.1 at 2026-05-08T07:02:03Z with exit code 1 after 5s
❌ ERROR: 'java run' for 3.4.1 failed with exit code 1.
FAILED[java run][3.4.1][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.4.1" ./gradlew clean javaTest -Pcoordinates="io.ktor:ktor-http-cio-jvm:3.2.0"]
```

--- a/metadata/io.ktor/ktor-http-cio-jvm/3.4.1/reachability-metadata.json
+++ b/metadata/io.ktor/ktor-http-cio-jvm/3.4.1/reachability-metadata.json
@@ -1,0 +1,40 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "io.ktor.utils.io.ByteChannel"
+      },
+      "type": "io.ktor.utils.io.ByteChannel"
+    },
+    {
+      "condition": {
+        "typeReached": "io.ktor.utils.io.pool.DefaultPool"
+      },
+      "type": "io.ktor.utils.io.pool.DefaultPool"
+    },
+    {
+      "condition": {
+        "typeReached": "io.ktor.http.cio.MultipartKt"
+      },
+      "type": "kotlinx.coroutines.channels.BufferedChannel"
+    },
+    {
+      "condition": {
+        "typeReached": "io.ktor.http.cio.MultipartKt"
+      },
+      "type": "kotlinx.coroutines.internal.ConcurrentLinkedListNode"
+    },
+    {
+      "condition": {
+        "typeReached": "io.ktor.http.cio.MultipartKt"
+      },
+      "type": "kotlinx.coroutines.internal.Segment"
+    },
+    {
+      "condition": {
+        "typeReached": "io.ktor.utils.io.ByteReadChannelOperationsKt"
+      },
+      "type": "kotlinx.io.RefCountingCopyTracker"
+    }
+  ]
+}

--- a/metadata/io.ktor/ktor-http-cio-jvm/index.json
+++ b/metadata/io.ktor/ktor-http-cio-jvm/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "3.4.1",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/io/ktor/ktor-http-cio-jvm/$version$/ktor-http-cio-jvm-$version$-sources.jar",
+    "repository-url" : "https://github.com/ktorio/ktor",
+    "test-code-url" : "https://github.com/ktorio/ktor/tree/$version$/ktor-http/ktor-http-cio/jvm/test",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/io/ktor/ktor-http-cio-jvm/$version$/ktor-http-cio-jvm-$version$-javadoc.jar",
+    "description" : "Ktor HTTP CIO provides the coroutine-based HTTP parsing, headers, chunked transfer, and multipart handling primitives used by Ktor's CIO networking stack on the JVM. It supplies low-level request and response builders and parsers that support Ktor client and server components built on Coroutine I/O.",
     "language" : {
       "name" : "kotlin",
       "version" : "2.1"

--- a/metadata/io.ktor/ktor-http-cio-jvm/index.json
+++ b/metadata/io.ktor/ktor-http-cio-jvm/index.json
@@ -1,38 +1,40 @@
 [
   {
-    "latest" : true,
-    "metadata-version" : "3.4.1",
-    "source-code-url" : "https://repo.maven.apache.org/maven2/io/ktor/ktor-http-cio-jvm/$version$/ktor-http-cio-jvm-$version$-sources.jar",
-    "repository-url" : "https://github.com/ktorio/ktor",
-    "test-code-url" : "https://github.com/ktorio/ktor/tree/$version$/ktor-http/ktor-http-cio/jvm/test",
-    "documentation-url" : "https://repo.maven.apache.org/maven2/io/ktor/ktor-http-cio-jvm/$version$/ktor-http-cio-jvm-$version$-javadoc.jar",
-    "description" : "Ktor HTTP CIO provides the coroutine-based HTTP parsing, headers, chunked transfer, and multipart handling primitives used by Ktor's CIO networking stack on the JVM. It supplies low-level request and response builders and parsers that support Ktor client and server components built on Coroutine I/O.",
-    "language" : {
-      "name" : "kotlin",
-      "version" : "2.1"
+    "latest": true,
+    "metadata-version": "3.4.1",
+    "source-code-url": "https://repo.maven.apache.org/maven2/io/ktor/ktor-http-cio-jvm/$version$/ktor-http-cio-jvm-$version$-sources.jar",
+    "repository-url": "https://github.com/ktorio/ktor",
+    "test-code-url": "https://github.com/ktorio/ktor/tree/$version$/ktor-http/ktor-http-cio/jvm/test",
+    "documentation-url": "https://repo.maven.apache.org/maven2/io/ktor/ktor-http-cio-jvm/$version$/ktor-http-cio-jvm-$version$-javadoc.jar",
+    "description": "Ktor HTTP CIO provides the coroutine-based HTTP parsing, headers, chunked transfer, and multipart handling primitives used by Ktor's CIO networking stack on the JVM. It supplies low-level request and response builders and parsers that support Ktor client and server components built on Coroutine I/O.",
+    "language": {
+      "name": "kotlin",
+      "version": "2.1"
     },
-    "tested-versions" : [
+    "tested-versions": [
       "3.4.1"
     ],
-    "allowed-packages" : [
-      "io.ktor.http.cio"
+    "allowed-packages": [
+      "io.ktor.http.cio",
+      "io.ktor.utils.io",
+      "io.ktor.utils.io.pool"
     ]
   },
   {
-    "metadata-version" : "3.2.0",
-    "source-code-url" : "https://repo.maven.apache.org/maven2/io/ktor/ktor-http-cio-jvm/$version$/ktor-http-cio-jvm-$version$-sources.jar",
-    "repository-url" : "https://github.com/ktorio/ktor",
-    "test-code-url" : "https://github.com/ktorio/ktor/tree/$version$/ktor-http/ktor-http-cio/jvm/test",
-    "documentation-url" : "https://repo.maven.apache.org/maven2/io/ktor/ktor-http-cio-jvm/$version$/ktor-http-cio-jvm-$version$-javadoc.jar",
-    "description" : "Ktor HTTP CIO provides the coroutine-based HTTP parsing, headers, chunked transfer, and multipart handling primitives used by Ktor's CIO networking stack on the JVM. It supplies low-level request and response builders and parsers that support Ktor client and server components built on Coroutine I/O.",
-    "language" : {
-      "name" : "kotlin",
-      "version" : "2.1"
+    "metadata-version": "3.2.0",
+    "source-code-url": "https://repo.maven.apache.org/maven2/io/ktor/ktor-http-cio-jvm/$version$/ktor-http-cio-jvm-$version$-sources.jar",
+    "repository-url": "https://github.com/ktorio/ktor",
+    "test-code-url": "https://github.com/ktorio/ktor/tree/$version$/ktor-http/ktor-http-cio/jvm/test",
+    "documentation-url": "https://repo.maven.apache.org/maven2/io/ktor/ktor-http-cio-jvm/$version$/ktor-http-cio-jvm-$version$-javadoc.jar",
+    "description": "Ktor HTTP CIO provides the coroutine-based HTTP parsing, headers, chunked transfer, and multipart handling primitives used by Ktor's CIO networking stack on the JVM. It supplies low-level request and response builders and parsers that support Ktor client and server components built on Coroutine I/O.",
+    "language": {
+      "name": "kotlin",
+      "version": "2.1"
     },
-    "tested-versions" : [
+    "tested-versions": [
       "3.2.0"
     ],
-    "allowed-packages" : [
+    "allowed-packages": [
       "io.ktor.http.cio"
     ]
   }

--- a/metadata/io.ktor/ktor-http-cio-jvm/index.json
+++ b/metadata/io.ktor/ktor-http-cio-jvm/index.json
@@ -1,6 +1,19 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "3.4.1",
+    "language" : {
+      "name" : "kotlin",
+      "version" : "2.1"
+    },
+    "tested-versions" : [
+      "3.4.1"
+    ],
+    "allowed-packages" : [
+      "io.ktor.http.cio"
+    ]
+  },
+  {
     "metadata-version" : "3.2.0",
     "source-code-url" : "https://repo.maven.apache.org/maven2/io/ktor/ktor-http-cio-jvm/$version$/ktor-http-cio-jvm-$version$-sources.jar",
     "repository-url" : "https://github.com/ktorio/ktor",

--- a/stats/io.ktor/ktor-http-cio-jvm/3.4.1/execution-metrics.json
+++ b/stats/io.ktor/ktor-http-cio-jvm/3.4.1/execution-metrics.json
@@ -1,0 +1,87 @@
+{
+  "fix_java_run_fail:2026-05-08": {
+    "timestamp": "2026-05-08T14:23:51.113342Z",
+    "library": "io.ktor:ktor-http-cio-jvm:3.4.1",
+    "starting_commit": "2837e776eaecbbe4fe344ab619247401900c829b",
+    "ending_commit": "895609063f518486e72d587a4e596f15afd315a6",
+    "previous_library": "io.ktor:ktor-http-cio-jvm:3.2.0",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "3.4.1",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 5382,
+          "missed": 2544,
+          "ratio": 0.679031,
+          "total": 7926
+        },
+        "line": {
+          "covered": 791,
+          "missed": 243,
+          "ratio": 0.76499,
+          "total": 1034
+        },
+        "method": {
+          "covered": 192,
+          "missed": 65,
+          "ratio": 0.747082,
+          "total": 257
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "3.2.0",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 4891,
+          "missed": 2005,
+          "ratio": 0.709252,
+          "total": 6896
+        },
+        "line": {
+          "covered": 786,
+          "missed": 235,
+          "ratio": 0.769833,
+          "total": 1021
+        },
+        "method": {
+          "covered": 199,
+          "missed": 62,
+          "ratio": 0.762452,
+          "total": 261
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 26403,
+      "cached_input_tokens_used": 168448,
+      "output_tokens_used": 1931,
+      "iterations": 1,
+      "cost_usd": 0.1421,
+      "tested_library_loc": 791,
+      "metadata_entries": 6,
+      "previous_library_metadata_entries": 0,
+      "code_coverage_percent": 76.5,
+      "previous_library_coverage_percent": 76.98
+    },
+    "artifacts": {
+      "test_file": "tests/src/io.ktor/ktor-http-cio-jvm/3.4.1/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt",
+      "metadata_file": "metadata/io.ktor/ktor-http-cio-jvm/3.4.1/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/io.ktor/ktor-http-cio-jvm/3.4.1/stats.json
+++ b/stats/io.ktor/ktor-http-cio-jvm/3.4.1/stats.json
@@ -1,0 +1,31 @@
+{
+  "versions" : [ {
+    "version" : "3.4.1",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 5382,
+        "missed" : 2544,
+        "ratio" : 0.679031,
+        "total" : 7926
+      },
+      "line" : {
+        "covered" : 791,
+        "missed" : 243,
+        "ratio" : 0.76499,
+        "total" : 1034
+      },
+      "method" : {
+        "covered" : 192,
+        "missed" : 65,
+        "ratio" : 0.747082,
+        "total" : 257
+      }
+    }
+  } ]
+}

--- a/tests/src/io.ktor/ktor-http-cio-jvm/3.4.1/.gitignore
+++ b/tests/src/io.ktor/ktor-http-cio-jvm/3.4.1/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/io.ktor/ktor-http-cio-jvm/3.4.1/build.gradle
+++ b/tests/src/io.ktor/ktor-http-cio-jvm/3.4.1/build.gradle
@@ -1,0 +1,36 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+    id "org.jetbrains.kotlin.jvm" version "2.3.20"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "io.ktor:ktor-http-cio-jvm:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+kotlin {
+    jvmToolchain(21)
+}
+
+compileTestKotlin {
+    kotlinOptions.jvmTarget = "21"
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/io.ktor/ktor-http-cio-jvm/3.4.1/gradle.properties
+++ b/tests/src/io.ktor/ktor-http-cio-jvm/3.4.1/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = io.ktor:ktor-http-cio-jvm:3.4.1
+library.version = 3.4.1
+metadata.dir = io.ktor/ktor-http-cio-jvm/3.4.1/

--- a/tests/src/io.ktor/ktor-http-cio-jvm/3.4.1/settings.gradle
+++ b/tests/src/io.ktor/ktor-http-cio-jvm/3.4.1/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'io.ktor.ktor-http-cio-jvm_tests'

--- a/tests/src/io.ktor/ktor-http-cio-jvm/3.4.1/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt
+++ b/tests/src/io.ktor/ktor-http-cio-jvm/3.4.1/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt
@@ -34,6 +34,7 @@ import kotlinx.io.Source
 import org.assertj.core.api.Assertions.assertThat
 import org.assertj.core.api.Assertions.assertThatThrownBy
 import org.junit.jupiter.api.Test
+import java.io.IOException
 import java.nio.ByteBuffer
 
 public class Ktor_http_cio_jvmTest {
@@ -212,7 +213,8 @@ public class Ktor_http_cio_jvmTest {
                 runBlocking {
                     decodeChunked(ByteReadChannel("1\r\na\r\nnot-a-size\r\n"), ByteChannel(autoFlush = true))
                 }
-            }.isInstanceOf(NumberFormatException::class.java)
+            }.isInstanceOf(IOException::class.java)
+                .hasMessageContaining("Invalid chunk size character")
         }
     }
 
@@ -293,7 +295,7 @@ public class Ktor_http_cio_jvmTest {
     }
 
     @Test
-    fun exposesParsedHeadersThroughIndexedAndOffsetAccessors(): Unit = runBlocking {
+    fun exposesParsedHeadersThroughOffsetAccessors(): Unit = runBlocking {
         withTimeout(10_000) {
             val headers = parseHeaders(
                 ByteReadChannel(
@@ -304,15 +306,12 @@ public class Ktor_http_cio_jvmTest {
                 )
             )
             try {
-                val indexedPairs: List<String> = (0 until headers.size)
-                    .map { "${headers.nameAt(it)}: ${headers.valueAt(it)}" }
                 val offsets: List<Int> = headers.offsets().toList()
                 val offsetPairs: List<String> = offsets
                     .map { "${headers.nameAtOffset(it)}: ${headers.valueAtOffset(it)}" }
 
                 assertThat(offsets).hasSize(headers.size)
-                assertThat(indexedPairs).containsExactlyInAnyOrder("Alpha: one", "Beta: two", "Alpha: three")
-                assertThat(offsetPairs).containsExactlyElementsOf(indexedPairs)
+                assertThat(offsetPairs).containsExactlyInAnyOrder("Alpha: one", "Beta: two", "Alpha: three")
             } finally {
                 headers.release()
             }

--- a/tests/src/io.ktor/ktor-http-cio-jvm/3.4.1/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt
+++ b/tests/src/io.ktor/ktor-http-cio-jvm/3.4.1/src/test/kotlin/io_ktor/ktor_http_cio_jvm/Ktor_http_cio_jvmTest.kt
@@ -1,0 +1,378 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package io_ktor.ktor_http_cio_jvm
+
+import io.ktor.http.HttpHeaders
+import io.ktor.http.HttpMethod
+import io.ktor.http.HttpProtocolVersion
+import io.ktor.http.cio.CIOHeaders
+import io.ktor.http.cio.ConnectionOptions
+import io.ktor.http.cio.MultipartEvent
+import io.ktor.http.cio.ParserException
+import io.ktor.http.cio.RequestResponseBuilder
+import io.ktor.http.cio.decodeChunked
+import io.ktor.http.cio.encodeChunked
+import io.ktor.http.cio.expectHttpBody
+import io.ktor.http.cio.expectHttpUpgrade
+import io.ktor.http.cio.parseHeaders
+import io.ktor.http.cio.parseHttpBody
+import io.ktor.http.cio.parseMultipart
+import io.ktor.http.cio.parseRequest
+import io.ktor.http.cio.parseResponse
+import io.ktor.utils.io.ByteChannel
+import io.ktor.utils.io.ByteReadChannel
+import io.ktor.utils.io.core.readText
+import io.ktor.utils.io.readRemaining
+import io.ktor.utils.io.toByteArray
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import kotlinx.io.Source
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.jupiter.api.Test
+import java.nio.ByteBuffer
+
+public class Ktor_http_cio_jvmTest {
+    @Test
+    fun parsesRequestsHeadersAndBodyExpectations(): Unit = runBlocking {
+        withTimeout(10_000) {
+            val rawRequest: String = """
+
+                GET /submit?name=alice HTTP/1.1
+                Host: example.com
+                Connection: keep-alive, Upgrade, X-Hop
+                Upgrade: websocket
+                Content-Length: 7
+                X-Custom: first
+                x-custom: second
+
+                payload-extra
+            """.trimIndent().replace("\n", "\r\n")
+            val channel = ByteReadChannel(rawRequest)
+            val request = parseRequest(channel) ?: error("Expected a request")
+
+            try {
+                assertThat(request.method).isEqualTo(HttpMethod.Get)
+                assertThat(request.uri.toString()).isEqualTo("/submit?name=alice")
+                assertThat(request.version.toString()).isEqualTo("HTTP/1.1")
+                assertThat(request.headers[HttpHeaders.Host].toString()).isEqualTo("example.com")
+                assertThat(request.headers.getAll("X-Custom").map { it.toString() }.toList())
+                    .containsExactly("first", "second")
+
+                val cioHeaders = CIOHeaders(request.headers)
+                assertThat(cioHeaders.caseInsensitiveName).isTrue()
+                assertThat(cioHeaders.get("x-custom")).isEqualTo("first")
+                assertThat(cioHeaders.getAll("X-CUSTOM")).containsExactly("first", "second")
+                assertThat(cioHeaders.contains("host", "example.com")).isTrue()
+                assertThat(cioHeaders.names()).contains("Host", "Connection", "Upgrade", "Content-Length", "X-Custom")
+
+                val connection = ConnectionOptions.parse(request.headers[HttpHeaders.Connection])
+                assertThat(connection).isEqualTo(
+                    ConnectionOptions(keepAlive = true, upgrade = true, extraOptions = listOf("X-Hop"))
+                )
+                assertThat(connection.toString()).isEqualTo("keep-alive, Upgrade, X-Hop")
+                assertThat(expectHttpUpgrade(request)).isTrue()
+                assertThat(expectHttpBody(request)).isTrue()
+
+                val body = ByteChannel(autoFlush = true)
+                parseHttpBody(request.headers, channel, body)
+                body.flushAndClose()
+                assertThat(body.toByteArray().decodeToString()).isEqualTo("payload")
+            } finally {
+                request.release()
+            }
+        }
+    }
+
+    @Test
+    fun parsesResponsesBuiltWithRequestResponseBuilder(): Unit = runBlocking {
+        withTimeout(10_000) {
+            val builder = RequestResponseBuilder()
+            try {
+                builder.responseLine("HTTP/1.1", 201, "Created")
+                builder.headerLine("Content-Type", "text/plain")
+                builder.headerLine("Set-Cookie", "a=1")
+                builder.headerLine("Set-Cookie", "b=2")
+                builder.emptyLine()
+                builder.bytes("accepted".encodeToByteArray())
+
+                val response = parseResponse(ByteReadChannel(builder.build())) ?: error("Expected a response")
+                try {
+                    assertThat(response.version.toString()).isEqualTo("HTTP/1.1")
+                    assertThat(response.status).isEqualTo(201)
+                    assertThat(response.statusText.toString()).isEqualTo("Created")
+                    assertThat(response.headers["content-type"].toString()).isEqualTo("text/plain")
+                    assertThat(response.headers.getAll("set-cookie").map { it.toString() }.toList())
+                        .containsExactly("a=1", "b=2")
+                } finally {
+                    response.release()
+                }
+            } finally {
+                builder.release()
+            }
+        }
+    }
+
+    @Test
+    fun requestResponseBuilderWritesLinesByteArraysAndByteBuffers(): Unit {
+        val builder = RequestResponseBuilder()
+        try {
+            builder.requestLine(HttpMethod.Put, "/resource/1", "HTTP/1.0")
+            builder.headerLine("Host", "localhost")
+            builder.line("X-Trace: abc")
+            builder.emptyLine()
+            builder.bytes("alpha".encodeToByteArray())
+            builder.bytes("0123456789".encodeToByteArray(), offset = 2, length = 4)
+            builder.bytes(ByteBuffer.wrap("omega".encodeToByteArray()))
+
+            val packet: Source = builder.build()
+            packet.use {
+                assertThat(it.readText()).isEqualTo(
+                    "PUT /resource/1 HTTP/1.0\r\n" +
+                        "Host: localhost\r\n" +
+                        "X-Trace: abc\r\n" +
+                        "\r\n" +
+                        "alpha2345omega"
+                )
+            }
+        } finally {
+            builder.release()
+        }
+    }
+
+    @Test
+    fun parsesStandaloneHeadersAndRejectsInvalidHeaderSyntax(): Unit = runBlocking {
+        withTimeout(10_000) {
+            val headers = parseHeaders(
+                ByteReadChannel(
+                    "Host: example.org\n" +
+                        "Transfer-Encoding: identity, chunked\n" +
+                        "Connection: close\n" +
+                        "X-Whitespace: \t value with trimmed padding   \n" +
+                        "\n"
+                )
+            )
+            try {
+                assertThat(headers.size).isEqualTo(4)
+                assertThat(headers["host"].toString()).isEqualTo("example.org")
+                assertThat(headers["X-Whitespace"].toString()).isEqualTo("value with trimmed padding")
+                assertThat(expectHttpBody(HttpMethod.Post, -1, headers["Transfer-Encoding"], null, null)).isTrue()
+                assertThat(expectHttpBody(HttpMethod.Get, -1, null, ConnectionOptions.Close, null)).isFalse()
+                assertThat(expectHttpBody(HttpMethod.Post, -1, null, ConnectionOptions.Close, null)).isTrue()
+                assertThat(expectHttpBody(HttpMethod.Head, 5, null, null, null)).isTrue()
+                assertThat(expectHttpUpgrade(HttpMethod.Post, "websocket", ConnectionOptions.Upgrade)).isFalse()
+            } finally {
+                headers.release()
+            }
+
+            assertThatThrownBy {
+                runBlocking {
+                    parseHeaders(ByteReadChannel("Host: bad/path\r\n\r\n"))
+                }
+            }.isInstanceOf(ParserException::class.java)
+
+            assertThatThrownBy {
+                runBlocking {
+                    parseHeaders(ByteReadChannel(": empty\r\n\r\n"))
+                }
+            }.isInstanceOf(ParserException::class.java)
+        }
+    }
+
+    @Test
+    fun encodesDecodesChunkedBodiesAndParsesChunkedHttpBody(): Unit = runBlocking {
+        withTimeout(10_000) {
+            val encodedOutput = ByteChannel(autoFlush = true)
+            encodeChunked(encodedOutput, ByteReadChannel("Wikipedia"))
+            encodedOutput.flushAndClose()
+            val encoded = encodedOutput.toByteArray().decodeToString()
+
+            assertThat(encoded).isEqualTo("9\r\nWikipedia\r\n0\r\n\r\n")
+
+            val decodedOutput = ByteChannel(autoFlush = true)
+            decodeChunked(ByteReadChannel("4\r\nWiki\r\n5\r\npedia\r\n0\r\n\r\n"), decodedOutput)
+            assertThat(decodedOutput.toByteArray().decodeToString()).isEqualTo("Wikipedia")
+
+            val parsedBody = ByteChannel(autoFlush = true)
+            parseHttpBody(
+                HttpProtocolVersion.HTTP_1_1,
+                contentLength = -1,
+                transferEncoding = "identity, chunked",
+                connectionOptions = ConnectionOptions.KeepAlive,
+                input = ByteReadChannel("5\r\nhello\r\n0\r\n\r\n"),
+                out = parsedBody
+            )
+            assertThat(parsedBody.toByteArray().decodeToString()).isEqualTo("hello")
+
+            assertThatThrownBy {
+                runBlocking {
+                    decodeChunked(ByteReadChannel("1\r\na\r\nnot-a-size\r\n"), ByteChannel(autoFlush = true))
+                }
+            }.isInstanceOf(NumberFormatException::class.java)
+        }
+    }
+
+    @Test
+    fun parsesMultipartEventsWithPreambleHeadersFilePartAndEpilogue(): Unit = runBlocking {
+        withTimeout(10_000) {
+            val boundary = "NativeBoundary"
+            val body = listOf(
+                "POST /upload HTTP/1.1",
+                "Host: upload.example",
+                "Content-Type: multipart/form-data; boundary=$boundary",
+                "Connection: close",
+                "",
+                "preamble text",
+                "--$boundary",
+                "Content-Disposition: form-data; name=\"field\"",
+                "X-Part: one",
+                "",
+                "field value",
+                "--$boundary",
+                "Content-Disposition: form-data; name=\"file\"; filename=\"notes.txt\"",
+                "Content-Type: text/plain",
+                "Content-Length: 11",
+                "",
+                "hello file!",
+                "--$boundary--",
+                "epilogue text"
+            ).joinToString("\r\n")
+
+            val input = ByteReadChannel(body)
+            val request = parseRequest(input) ?: error("Expected multipart request")
+            try {
+                val events = parseMultipart(input, request.headers)
+
+                val preamble = events.receive() as MultipartEvent.Preamble
+                try {
+                    assertThat(preamble.body.readText()).isEqualTo("preamble text\r\n")
+                } finally {
+                    preamble.release()
+                }
+
+                val field = events.receive() as MultipartEvent.MultipartPart
+                val fieldHeaders = field.headers.await()
+                try {
+                    assertThat(fieldHeaders["Content-Disposition"].toString())
+                        .isEqualTo("form-data; name=\"field\"")
+                    assertThat(fieldHeaders["x-part"].toString()).isEqualTo("one")
+                    assertThat(field.body.readRemaining().readText()).isEqualTo("field value")
+                } finally {
+                    fieldHeaders.release()
+                    field.release()
+                }
+
+                val file = events.receive() as MultipartEvent.MultipartPart
+                val fileHeaders = file.headers.await()
+                try {
+                    val headers = CIOHeaders(fileHeaders)
+                    assertThat(headers.get("content-type")).isEqualTo("text/plain")
+                    assertThat(headers.get("content-length")).isEqualTo("11")
+                    assertThat(file.body.readRemaining().readText()).isEqualTo("hello file!")
+                } finally {
+                    fileHeaders.release()
+                    file.release()
+                }
+
+                val epilogue = events.receive() as MultipartEvent.Epilogue
+                try {
+                    assertThat(epilogue.body.readText()).isEqualTo("epilogue text")
+                } finally {
+                    epilogue.release()
+                }
+
+                assertThat(events.receiveCatching().isClosed).isTrue()
+            } finally {
+                request.release()
+            }
+        }
+    }
+
+    @Test
+    fun exposesParsedHeadersThroughIndexedAndOffsetAccessors(): Unit = runBlocking {
+        withTimeout(10_000) {
+            val headers = parseHeaders(
+                ByteReadChannel(
+                    "Alpha: one\r\n" +
+                        "Beta: two\r\n" +
+                        "Alpha: three\r\n" +
+                        "\r\n"
+                )
+            )
+            try {
+                val indexedPairs: List<String> = (0 until headers.size)
+                    .map { "${headers.nameAt(it)}: ${headers.valueAt(it)}" }
+                val offsets: List<Int> = headers.offsets().toList()
+                val offsetPairs: List<String> = offsets
+                    .map { "${headers.nameAtOffset(it)}: ${headers.valueAtOffset(it)}" }
+
+                assertThat(offsets).hasSize(headers.size)
+                assertThat(indexedPairs).containsExactlyInAnyOrder("Alpha: one", "Beta: two", "Alpha: three")
+                assertThat(offsetPairs).containsExactlyElementsOf(indexedPairs)
+            } finally {
+                headers.release()
+            }
+        }
+    }
+
+    @Test
+    fun parsesBodyDelimitedByConnectionClose(): Unit = runBlocking {
+        withTimeout(10_000) {
+            val output = ByteChannel(autoFlush = true)
+            parseHttpBody(
+                HttpProtocolVersion.HTTP_1_1,
+                contentLength = -1,
+                transferEncoding = null,
+                connectionOptions = ConnectionOptions.Close,
+                input = ByteReadChannel("streamed until close"),
+                out = output
+            )
+            output.flushAndClose()
+
+            assertThat(output.toByteArray().decodeToString()).isEqualTo("streamed until close")
+        }
+    }
+
+    @Test
+    fun rejectsMalformedRequestsResponsesAndUnsupportedBodyEncodings(): Unit = runBlocking {
+        withTimeout(10_000) {
+            assertThat(parseRequest(ByteReadChannel("\r\n\r\n"))).isNull()
+            assertThat(parseResponse(ByteReadChannel(""))).isNull()
+
+            assertThatThrownBy {
+                runBlocking {
+                    parseRequest(ByteReadChannel("GET / HTTP/2.0\r\n\r\n"))
+                }
+            }.isInstanceOf(ParserException::class.java)
+                .hasMessageContaining("Unsupported HTTP version")
+
+            assertThatThrownBy {
+                runBlocking {
+                    parseResponse(ByteReadChannel("HTTP/1.1 99 TooLow\r\n\r\n"))
+                }
+            }.isInstanceOf(ParserException::class.java)
+                .hasMessageContaining("Status-code must be 3-digit")
+
+            assertThatThrownBy {
+                expectHttpBody(HttpMethod.Post, -1, "gzip", null, "text/plain")
+            }.isInstanceOf(IllegalArgumentException::class.java)
+                .hasMessageContaining("Unsupported transfer encoding")
+
+            val noLengthOutput = ByteChannel(autoFlush = true)
+            parseHttpBody(
+                HttpProtocolVersion.HTTP_1_1,
+                contentLength = -1,
+                transferEncoding = null,
+                connectionOptions = ConnectionOptions.KeepAlive,
+                input = ByteReadChannel("body"),
+                out = noLengthOutput
+            )
+            assertThat(noLengthOutput.closedCause)
+                .hasMessageContaining("Failed to parse request body")
+        }
+    }
+}

--- a/tests/src/io.ktor/ktor-http-cio-jvm/3.4.1/user-code-filter.json
+++ b/tests/src/io.ktor/ktor-http-cio-jvm/3.4.1/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "io.ktor.http.cio.**"
+    },
+    {
+      "includeClasses" : "io_ktor.ktor_http_cio_jvm.**"
+    }
+  ]
+}

